editor/modes/visual-editor/block.js

```diff
@@ -133,6 +134,10 @@ class VisualEditorBlock extends wp.element.Component {
 
 		const { onSelect, onStartTyping, onHover, onMouseLeave, onFocus, onInsertAfter } = this.props;
 
+		const toolbars = settings.controls && settings.controls.length && ! Array.isArray( first( settings.controls ) )
```

Comment:
>if we have mixed values, the controls outside arrays are included in the previous array, which I think is not the desired behaviour if we ever have mixed controls

I don't know that we'd want to encourage this anyways, but in the current implementation this will simply explode (trying to `map` on a non-array); thus the "graceful" remark.